src/libexpr/value/print.hh

```diff
@@ -0,0 +1,30 @@
+#pragma once
+/**
+ * @file
+ * @brief Common printing functions for the Nix language
+ *
+ * While most types come with their own methods for printing, they share some
+ * functions that are placed here.
+ */
+
+#include <iostream>
+
+namespace nix {
+    /**
+     * Print a string as a Nix string literal.
+     *
+     * Quotes and fairly minimal escaping are added.
+     *
+     * @param s The logical string
+     */
+    std::ostream & printLiteral(std::ostream & o, std::string_view s);
```

Comment:
I'm not sure if this overloaded `printLiteral` function (as opposed to `printBool`, `printString` etc.) is a good idea. For instance, at some point somebody is going to do `printLiteral(null)`, expecting a `null` value to be printed, but instead it will call the `char *` overload...

Reply:
Done. It's `printLiteralString` now, so that it's explicit which kind of literal will be rendered; same for `printLiteralBool`.
I've kept the "literal" part because I've renamed the file to cover relevant parts of the whole language, not just the values.